From fd26354985c824dc1de36a556a040a294f7f1515 Mon Sep 17 00:00:00 2001
From: Zhang Jiejing <jiejing.zhang@freescale.com>
Date: Wed, 18 Jul 2012 17:03:09 +0800
Subject: [PATCH 311/498] ENGR00217401-3 gralloc: fix type error of phy memory
 address.

fix the phy memory adderss type, should be u32 rather that signed.

Signed-off-by: Zhang Jiejing <jiejing.zhang@freescale.com>
---
 mx6/libcamera/CameraHal.cpp           | 2 +-
 mx6/libgralloc_wrapper/gralloc_priv.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/mx6/libcamera/CameraHal.cpp b/mx6/libcamera/CameraHal.cpp
index db5e016..06099bc 100755
--- a/mx6/libcamera/CameraHal.cpp
+++ b/mx6/libcamera/CameraHal.cpp
@@ -831,7 +831,7 @@ namespace android {
             private_handle_t *handle = (private_handle_t *)(*buf_h);
             mapper.lock(handle, GRALLOC_USAGE_SW_READ_OFTEN | GRALLOC_USAGE_SW_WRITE_OFTEN, bounds, &pVaddr);
             if((handle->phys == 0) || (handle->base == 0) || (handle->size == 0)) {
-                 CAMERA_LOG_ERR("%s: dequeue invalide Buffer, phys=%x, base=%x, size=%d", __FUNCTION__, handle->phys, handle->base, handle->size);
+                 CAMERA_LOG_ERR("%s: dequeue invalide Buffer, phys=0x%lx, base=%x, size=%d", __FUNCTION__, handle->phys, handle->base, handle->size);
                  mNativeWindow->cancel_buffer(mNativeWindow, buf_h);
                  return BAD_VALUE;
             }
diff --git a/mx6/libgralloc_wrapper/gralloc_priv.h b/mx6/libgralloc_wrapper/gralloc_priv.h
index 13dbc1a..304bca7 100755
--- a/mx6/libgralloc_wrapper/gralloc_priv.h
+++ b/mx6/libgralloc_wrapper/gralloc_priv.h
@@ -93,7 +93,7 @@ struct private_handle_t {
     int     size;
     int     offset;
     int     base;
-    int     phys;
+    unsigned long phys;
     int     format;
     int     width;
     int     height;
-- 
1.8.0

